waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/impl/PrefixBasedDatabaseMappingService.java

```diff
@@ -92,6 +92,12 @@ public PrefixBasedDatabaseMappingService(
 
   private void add(AbstractMetaStore metaStore) {
     MetaStoreMapping metaStoreMapping = metaStoreMappingFactory.newInstance(metaStore);
+    // connect metastore first
+    if (!metaStoreMapping.isAvailable()) {
```

Comment:
why was this added?

Reply:
Hmm not sure I follow, this method just adds a Metastore to the list of configured metastores and you're just logging a warning. 
You've already changed the code to do the delegate connection here:
[DefaultMetaStoreClientFactory.java](https://github.com/ExpediaGroup/waggle-dance/pull/266/files#diff-ee675b8d870d29e0855316cdace45011de4a0a97a308fe7cefbabb4937baa63a)
Which is the correct place to add the logic. 

Adding this here just forces a client.isOpen/getStatus call earlier in the process. Might not be needed...

Reply:
The call `StaticDatabaseMappingService` is done because we have to get the database e.g. make a real connection.
Here I think we can defer till we get a use call (Which will do the getDelegate then) but it might not have a lot of impact. We can leave it an see.

Reply:
Hi been testing with this call and please remove this. It will impact normal calls.
Considers federating to 10+ metastores we'll be doing this isAvailable check for all of them and if any of those are slow to respond we've added unnecessary latency. These calls should happen lazily only if needed. 
Which I think will work ok because of your changes in the `DefaultMetaStoreClientFactory`.

Reply:
This part of the code should not have any knowledge of SASL or non SASL connection it should treat it all the same.

If this call is needed for Kerberos then something might need to be added in: `TProcessorFactorySaslDecorator`
Specifically I am thinking that perhaps you can inject: `NotifyingFederationService notifyingFederationService` and ` MetaStoreMappingFactory metaStoreMappingFactory` in the `MetaStoreProxyServer`

Then add it to call here: https://github.com/ExpediaGroup/waggle-dance/pull/266/files#diff-b9afe479963c18cd3af539fa88f9b7b728510cf1ad2b24a2ee26658a423770c1R210

In the `TProcessorFactorySaslDecorator.getProcess()`
Do a loop over notifyingFederationService.getAll() for each create a mapping and check availability. That way the extra work is only done for kerberos and won't affect anything else.